data/basal.go

```diff
@@ -0,0 +1,166 @@
+package data
+
+import (
+	"fmt"
+	"reflect"
+	"strings"
+
+	valid "github.com/tidepool-org/platform/Godeps/_workspace/src/gopkg.in/bluesuncorp/validator.v8"
+	"github.com/tidepool-org/platform/validate"
+)
+
+func init() {
+	validator.RegisterValidation(rateTag, BasalRateValidator)
+	validator.RegisterValidation(durationTag, BasalDurationValidator)
+	validator.RegisterValidation(deliveryTypeTag, BasalDeliveryTypeValidator)
+	validator.RegisterValidation(insulinTag, BasalInsulinValidator)
+	validator.RegisterValidation(valueTag, BasalValueValidator)
+}
+
+//Basal represents a basal device data record
+type Basal struct {
+	DeliveryType string          `json:"deliveryType" bson:"deliveryType" valid:"basaldeliverytype"`
+	ScheduleName string          `json:"scheduleName" bson:"scheduleName" valid:"omitempty,required"`
+	Rate         float64         `json:"rate" bson:"rate" valid:"omitempty,basalrate"`
+	Duration     int             `json:"duration" bson:"duration" valid:"omitempty,basalduration"`
+	Insulin      string          `json:"insulin" bson:"insulin,omitempty" valid:"omitempty,basalinsulin"`
+	Value        int             `json:"value" bson:"value,omitempty" valid:"omitempty,basalvalue"`
+	Suppressed   *SupressedBasal `json:"suppressed" bson:"suppressed,omitempty" valid:"omitempty,required"`
+	Base         `bson:",inline"`
+}
+
+//SupressedBasal represents a suppressed basal portion of a basal
+type SupressedBasal struct {
+	Type         string  `json:"type" bson:"type" valid:"required"`
+	DeliveryType string  `json:"deliveryType" bson:"deliveryType" valid:"basaldeliverytype"`
+	ScheduleName string  `json:"scheduleName" bson:"scheduleName" valid:"omitempty,required"`
+	Rate         float64 `json:"rate" bson:"rate" valid:"omitempty,basalrate"`
+}
+
+const (
+	//BasalName is the given name for the type of a `Basal` datum
+	BasalName = "basal"
+
+	deliveryTypeField = "deliveryType"
+	scheduleNameField = "scheduleName"
+	insulinField      = "insulin"
+	valueField        = "value"
+	rateField         = "rate"
+	durationField     = "duration"
+
+	deliveryTypeTag validate.ValidationTag = "basaldeliverytype"
```

Comment:
Should we use camelCase values here?  `basalDeliveryType`?  We use that for our json and bson tags.  Or is there a limitation with validator?